pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/recipe/RecipeTypes.kt

```diff
@@ -83,8 +85,10 @@ private object SmithingRecipeType : VanillaRecipe<SmithingRecipe>("smithing") {
     @EventHandler(priority = EventPriority.LOWEST)
     private fun onSmith(e: PrepareSmithingEvent) {
         val inv = e.inventory
+        val recipe = inv.recipe
+        if(recipe == null || recipe !is Keyed) return
```

Comment:
```suggestion
        if(recipe !is Keyed) return
```

Not blocking but the `is` checks nullable as well

Reply:
did for both